src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetFeatureInfoTest.java

```diff
@@ -960,4 +966,31 @@ public void testNotQueryableGroupLayer() throws Exception {
        
        catalog.remove(layerGroup);
    }
+   
+   @Test
+   public void testGetFeatureInfoOpaqueGroup() throws Exception {
+       String url = "wms?LAYERS=" + OPAQUE_GROUP + "&STYLES=&FORMAT=image%2Fpng"
+               + "&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetFeatureInfo&SRS=EPSG%3A4326&WIDTH=256&HEIGHT=256&BBOX=-0.0043,-0.0025,0.0043,0.0025" +
+               "&info_format=text/plain&request=GetFeatureInfo&&query_layers=" + OPAQUE_GROUP + "&x=105&y=107";
+       String response = getAsString(url);
+       assertThat(response, containsString("FID = 102"));
+   }
+
+   @Test
+   public void testFeatureInfoLayersInOpaqueGroup() throws Exception {
+       LayerGroupInfo group = getCatalog().getLayerGroupByName(OPAQUE_GROUP);
+       for (PublishedInfo pi : group.layers()) {
+           final String layerName = pi.prefixedName();
+        String url = "wms?LAYERS=" + layerName + "&STYLES=&FORMAT=image%2Fpng"
+                   + "&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetFeatureInfo&SRS=EPSG%3A4326&WIDTH=256&HEIGHT=256&BBOX=-0.0043,-0.0025,0.0043,0.0025" +
+                   "&info_format=text/plain&request=GetFeatureInfo&&query_layers=" + layerName + "&x=105&y=107";
+           Document dom = getAsDOM(url);
+           //print(dom);
+           
+           // should not be found
+           XMLAssert.assertXpathEvaluatesTo("1", "count(/ServiceExceptionReport)", dom);
+           XMLAssert.assertXpathEvaluatesTo("layers", "//ServiceException/@locator", dom);
+           XMLAssert.assertXpathEvaluatesTo("LayerNotDefined", "//ServiceException/@code", dom);
+       }
+   }
 }
```

Comment:
Manual testing shows that layers contained in an opaque container group are actually queryable, so wondering how this test can succeeded :astonished:.